src/net.cpp

```diff
@@ -2470,28 +2473,31 @@ bool CBanDB::Read(banmap_t& banSet)
         // ... verify the network matches ours
         if (memcmp(pchMsgTmp, Params().MessageStart(), sizeof(pchMsgTmp)))
             return error("%s: Invalid network magic number", __func__);
-        
+
         // de-serialize address data into one CAddrMan object
         ssBanlist >> banSet;
     }
     catch (const std::exception& e) {
         return error("%s: Deserialize or I/O error - %s", __func__, e.what());
     }
-    
+
     return true;
 }
 
 void DumpBanlist()
 {
-    int64_t nStart = GetTimeMillis();
+    CNode::SweepBanned(); // clean unused entires (if bantime has expired)
 
-    CNode::SweepBanned(); //clean unused entries (if bantime has expired)
+    if (CNode::BannedSetIsDirty()) {
+        int64_t nStart = GetTimeMillis();
 
-    CBanDB bandb;
-    banmap_t banmap;
-    CNode::GetBanned(banmap);
-    bandb.Write(banmap);
+        CBanDB bandb;
+        banmap_t banmap;
+        CNode::GetBanned(banmap);
+        if (bandb.Write(banmap))
+            CNode::SetBannedSetDirty(false);
 
-    LogPrint("net", "Flushed %d banned node ips/subnets to banlist.dat  %dms\n",
-             banmap.size(), GetTimeMillis() - nStart);
+        LogPrint("net", "Flushed %d banned node ips/subnets to banlist.dat  %dms\n",
+            banmap.size(), GetTimeMillis() - nStart);
+    }
```

Comment:
General principle:  Code like "if (!dirty) return;" reduces the indentation level for the standard case, and makes the code more readable.